src/control/cmd/dmg/storage.go

```diff
@@ -155,48 +155,79 @@ type storageFormatCmd struct {
 	ctlInvokerCmd
 	hostListCmd
 	jsonOutputCmd
-	Verbose  bool `short:"v" long:"verbose" description:"Show results of each SCM & NVMe device format operation"`
-	Reformat bool `long:"reformat" description:"Always reformat storage (CAUTION: Potentially destructive)"`
-	Group    struct {
-		System bool   `long:"system" description:"Perform reformat of stopped DAOS servers in system"`
-		Ranks  string `long:"ranks" short:"r" description:"Comma separated list of system ranks to format, default is all ranks"`
-	} `group:"System Format Options" description:"Reformat an existing DAOS system"`
+	Verbose  bool   `short:"v" long:"verbose" description:"Show results of each SCM & NVMe device format operation"`
+	Reformat bool   `long:"reformat" description:"Reformat storage overwriting any existing filesystem (CAUTION: Potentially destructive)"`
+	Ranks    string `long:"ranks" short:"r" description:"Comma separated list of system ranks to format, default is all ranks"`
 }
 
-// Execute is run when storageFormatCmd activates
+// shouldReformatSystem queries system to interrogate membership before deciding
+// whether a system reformat is appropriate.
 //
-// run NVMe and SCM storage format on all connected servers
-func (cmd *storageFormatCmd) Execute(args []string) (err error) {
-	ctx := context.Background()
+// Only reformat system if membership is not empty and all ranks are stopped.
+func (cmd *storageFormatCmd) shouldReformatSystem(ctx context.Context, ranks []system.Rank) (bool, error) {
+	if cmd.Reformat {
+		cmd.log.Info("processing system reformat request")
 
-	if cmd.Group.System {
-		cmd.log.Info("system reformat selected")
+		resp, err := control.SystemQuery(ctx, cmd.ctlInvoker, &control.SystemQueryReq{})
+		if err != nil {
+			return false, errors.Wrap(err, "System-Query command failed")
+		}
 
-		if cmd.Reformat {
-			cmd.log.Info("--reformat is already implied by --system")
+		if len(resp.Members) == 0 {
+			cmd.log.Debug("no system members, reformat host list")
+			if len(ranks) > 0 {
+				cmd.log.Info("--ranks parameter will be ignored as membership is empty")
+			}
+
+			return false, nil
 		}
 
-		ranks, err := system.ParseRanks(cmd.Group.Ranks)
-		if err != nil {
-			return errors.Wrap(err, "parsing rank list")
+		for _, member := range resp.Members {
+			if member.State() != system.MemberStateStopped {
+				return false, errors.Errorf(
+					"system reformat requires all system ranks to be stopped (%+v)",
+					member)
+			}
```

Comment:
Might be better here to collect all ranks that are in a non-stopped state and display them as a ranklist in the error message. It's true that the admin can also learn this information by using `dmg system query`, but if they're going to have to do it anyhow I think we might as well just do it here and remove the need to do another query.